src/v/model/ktp.h

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
```

Comment:
I think it's debatable whether this should be in its own header. Arguably it could be in fundamental.h.

Having it separate during development was very nice since I could avoid recompiling the world (everything that includes fundamental.h) but open to suggestions about whether it should stay like that in the final version.

Reply:
Yes I don't think small headers really have any penalty on compile time at all, it was more just if it made sense logically.

Reply:
Cool, leaving it as-is.